dvc/fs/path.py

```diff
@@ -10,31 +11,15 @@ def __init__(self, sep):
             self.flavour = ntpath
         else:
             raise ValueError(f"unsupported separator '{sep}'")
+        self.sep_re = re.compile(f"{re.escape(sep)}+")
 
     def join(self, *parts):
         return self.flavour.join(*parts)
 
     def parts(self, path):
-        drive, path = self.flavour.splitdrive(path)
-
-        ret = []
-        while True:
-            path, part = self.flavour.split(path)
-
-            if part:
-                ret.append(part)
-                continue
-
-            if path:
-                ret.append(path)
-
-            break
-
-        ret.reverse()
-
-        if drive:
-            ret = [drive] + ret
-
+        ret = self.sep_re.split(path.strip(self.flavour.sep))
+        if path.startswith(self.flavour.sep):
+            ret = [self.flavour.sep] + ret
```

Comment:
and looks like it broke windows tests 🙁 Looks like the minimal change is to just rstrip the sep from the path, right?

Reply:
Yeah, let's go with the minimal fix for now. I don't want to spend a lot of time debugging the windows tests now. I made a separate PR since I still think this might be worth debugging and working off of at some point. Let me know if you don't think so, but I think the current implementation still doesn't work as expected for windows paths.

Here's a PR with the minimal fix. Looks like it passes the tests: https://github.com/iterative/dvc/pull/7234